### DISCUSSION TITLE
Capital recovery factor calucation

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#92] [Q&A: Analysis]
Hi ReEDS team!

I have a question about the calculation of the capital recovery factor. In the "Regional Energy Deployment System (ReEDS) Model Document: 2020 Edition". It describes the capital recovery factor as equal to (1- 1/(1+ real discount rate)^economic horizon)/real discount rate. However, in support_function.py, the formula is calculated as sys_financials['crf'] = (sys_financials['d_real'] - 1. 0) / (1 - (1 / sys_financials['d_real']**sys _eval_years )), which stands for: real discount rate / (1 - 1/(1 + real discount rate) ^ economic horizon). Am I understanding this correctly? If correct, why the crf equation in python is different from the one in the report file.

Attached is a screenshot of the report 
<img width="472" alt="CRF_eq_tech-doc" src="https://user-images.githubusercontent.com/78582696/153156698-de9c24a7-79c6-445f-8f51-09fe017a5e0a.PNG">
and a screenshot of the python file .
<img width="703" alt="CRF_eq_python" src="https://user-images.githubusercontent.com/78582696/153156693-48dbf637-52c3-4618-9dc6-e27ac462aafd.PNG">

Thanks!

**Comment by wesleyjcole** [comment]
What we have in Python is the reciprocal of the CRF in the report.  So we've basically calculated the 1/CRF value and call it the CRF in the code.

**Reply by wanyingw193** [reply]
Got it! Does this mean that the crf.csv output from calc_financial_inputs.py using import_sys_financials from support_function.py is also the reciprocal of the CRF in the report?

In ReEDS d_solveprep.gms, pvf_onm(t) is the result of 1/crf(t), where crf(t) is the crf.csv from python. Is pvf_onm(t) in objective function used to calculate "present value of annuity" or "annuity"? Thank you so much!

**Reply by wesleyjcole** [reply]
The report is incorrect in missing the reciprocal:

Present value of Annuity = Annuity * (1- 1/(1+d_r)^lifetime)/d_r

So it is actually:

Present value of Annuity / Annuity = (1- 1/(1+d_r)^lifetime)/d_r = 1 / CRF

See also https://en.wikipedia.org/wiki/Capital_recovery_factor.

The Python code has the correct handling, and it looks like we need to update the documentation to reflect this better.

**Reply by wanyingw193** [reply]
Got it! Thanks for the explanation!

**Comment by wanyingw193** [comment]
Hi, sorry, I have two more simple questions about pvf_capital(t).
Can I understand it in this way:

- The capital cost of a new asset is only added to the objective function in its first installation year. 
- The pvf_capital(t) is 1 because the crf of the capital cost is already calculated in finance_risk_mult in support_funcation.py.

![image](https://user-images.githubusercontent.com/78582696/153289189-59b65145-2579-4130-b5a1-3bf7ad8c88b3.png)

**Reply by wesleyjcole** [reply]
In sequential mode, the pvf_capital is 1 because we evaluate each year independently (e.g., when solving for 2030 you only worry about builds that occur in 2030).  In intertemporal mode pvf_capital varies to capture the difference in value when investing in a plant in different years.  For example, the model would evaluate the trade-off between building a new PV plant in 2028 vs. 2030, and this pvf_capital would inform the model about the difference in cost by building in 2028 vs. 2030 from a net present value perspective.

**Reply by wanyingw193** [reply]
Thank you!

**Comment by wanyingw193** [comment]
Hi ReEDS team, would you mind explaining the application of 'pvf_onm=1/capital recovery factor' in the objective function. What is the implication of choosing the capital recovery factor to calculate the present value of operating costs? 

I still can't fully understand it. Because in engineering economics, the capital recovery factor is used in the annualized calculation of capital cost. 

Thank you very much!

**Reply by wesleyjcole** [reply]
We do not annualize the capital costs in the objective function, so in order to have comparable operating costs we need to collapse the operating costs over the 20-year economic period into one value.  We do that using pvf_onm = 1/CRF.  This allows the operating costs to be compared with the capital costs when making investment decisions.

**Reply by wanyingw193** [reply]
Thank you for your explanation! I have better idea about the application of Pvf_onm now!

In support_function.py, the capital recovery factor is also applied in the calculation of df_inv['financing_risk_mult']. 

Here,  df_inv['financing_risk_mult']= 1.0 + df_inv['finance_diff_real'] * ( (1-(1/df_inv['d_real']**df_inv['eval_period']))/(df_inv['d_real']-1.0)). 

Where,  df_inv['finance_diff_real'] represent difference between system average and tech WACC; 
            ( (1-(1/df_inv['d_real']**df_inv['eval_period']))/(df_inv['d_real']-1.0)) represent 1/capital recovery factor. 
Am I understand it correctly?

Could you explain the implications of using the capital recovery factor in this equation? And where is the source of the data for df_inv['finance_diff_real']? 

Thank you and have a nice day!

**Reply by wesleyjcole** [reply]
The CRF is used for system cost calculations, but some technologies might have financing costs that are higher or lower than the system-wide financing costs.  The difference term calculates the differences between the system cost and the technology-specific cost.  The technology-specific costs are taken from the Annual Technology Baseline, which has financing costs for each technology.

**Reply by wanyingw193** [reply]
Got it! It is quite straightforward now. 

Based on the WACC calculation from support_function.py (d_real-1 = discount rate= system WACC). 
![image](https://user-images.githubusercontent.com/78582696/154321372-7261d716-a975-45a8-b307-ba5f507698fc.png)
I got the system WACC below, 
![image](https://user-images.githubusercontent.com/78582696/154321684-722d1364-b9b5-4c51-b984-f13017d65cfd.png)

Then, on ATB_2021, the nominal technology WACC for coal FE in 2019 is 6.5%, which is consistent with the 3% (i.e., 6.5%-3.5%) finance_diff_real in financials_tech_ATB2021.csv. 
![image](https://user-images.githubusercontent.com/78582696/154324277-691a741b-d845-48f5-9aa5-005fa5d36ded.png)
![image](https://user-images.githubusercontent.com/78582696/154324444-0c4e09ad-d1f6-4a5d-a946-d422670fc287.png)

However, the nominal technology WACC for CSP is 5.3%, so its WACC difference is 1.8%, and that number is different from the CSV file. Am I missing something to consider? Also, the WACC difference for the natural gas plant is 0%, while its nominal tech WACC from ATB_2021 is 6.5%. What is the assumption behind this number?

**Reply by wesleyjcole** [reply]
The finance_diff_real term is the difference from the nominal natural gas WACC and the technology's WACC using a 20-year cost recovery period and market factor financials.  We also did some syncing up of debt fractions during years where it is changing to make the values more comparable.

**Reply by wanyingw193** [reply]
Hi! Just wondering why the finance_diff_real of CCS and DAC was set to be 0?
![image](https://user-images.githubusercontent.com/78582696/178373049-6f0fbf8d-cf06-49ff-97c4-a434be0b1d8d.png)

**Reply by wesleyjcole** [reply]
There isn't data on coal and CCS financing costs, so as a starting point we set their values to zero.

**Reply by wanyingw193** [reply]
Got it! Thank you!

**Comment by wanyingw193** [comment]
Hi! I just wondering why ReEDS would choose the capital recovery factor to calculate the present value instead of using the discount factor (1+r)^-1. Thank you in advance!

**Comment by maxxb77** [comment]
It really depends on your perspective with the CRF being used as an
adjustment to operations to allow for myopic solve (ie weight operations as
if they were to persist for X years) and the discount factor being used to
create a NPV from the stream of costs across time. For ReEDS, the discount
factor is applied to the objective function in combination with the CRF
with the former being applied to both operations and investment and the
reciprocal of the latter being applied solely to the operations component.

Good question btw

**Comment by wanyingw193** [comment]
Hi, I just want to confirm some of my understanding.

As I understand it, pvf_onm is based on the constant t=20 to calculate the NPV operating cost in the objective function. And there are no other parameters/variables to capture the difference in plant age between existing and new installed capacity.

Or am I not fully understanding? Does the model have other parameters/variables to handle the different plant ages between existing and old capacity in the objective function?

As an example, if one plant is installed in 2010 and the second plant is installed in 2020. Then in 2024, first plant is 14 years old and the second plant is 4 years old, will they both be evaluated by collapsing the operating costs over a 20 year economic period? Or would the first plant's OM cost be collapsed over 6 years and the second plant over 16 years?

Thank you!

**Comment by maxxb77** [comment]
Responses to...

"As I understand it, pvf_onm is based on the constant t=20 to calculate the
NPV operating cost in the objective function."
- Correct, it weights operations as 20 years generally assuming that the
current conditions remain constant (although there are some means by which
we can say "fuel prices will rise" or "there's a penalty given foreseen
increasing policy stringency")

"And there are no other parameters/variables to capture the difference in
plant age between existing and new installed capacity."
- Not sure here, need more details to be 100% certain -but- there are
parameters like cost_cap_fin_mult which include assumed construction times,
loan lifetimes, incentives, and several other considerations. Generally,
tho, investments are considered fixed after they occur and the cost of
financing is faced upfront (ie rolled into an overnight cost)

"As an example, if one plant is installed in 2010 and the second plant is
installed in 2020. Then in 2024, first plant is 14 years old and the second
plant is 4 years old, will they both be evaluated by collapsing the
operating costs over a 20 year economic period? Or would the first plant's
OM cost be collapsed over 6 years and the second plant over 16 years?"

- In the sequential case, this is true... in intertemporal/window settings
the model can see that the plant has a finite lifespan (ie I'd see my
investment will only survive for X years). The myopic viewpoint and
assumption boils down to the question: would a generating unit behave
differently knowing that it will only survive for X more years? again
speaking generally, the perspective taken with ReEDS is that it would
generate regardless of remaining lifespan if the marginal revenue exceeds
marginal cost

--

If you'd like, you can reach out to me at maxwell.brown 'at' nrel.gov to
set up a meeting and go over some of these questions.. I'd also be
interested in hearing about your work

Best,
Max

**Reply by wanyingw193** [reply]
It will be great. I will digest the response for a bit and then reach out. Thank you so much Maxwell!

**Comment by wanyingw193** [comment]
Morning! I just want to know what are the equations for pvf_capital and pvf_onm in the intertemporal model? Do they use CRF or discount factor to calculate? When I did the CRF calculation, I can't get the same numbers from input_case/pvf_cap and pvf_onm_int. Could I get some guidance?

![image](https://user-images.githubusercontent.com/78582696/167082011-95fd5337-f8da-41dc-a1aa-e139f9cacff2.png)

**Comment by maxxb77** [comment]
So pvf_onm in intertemporal mode will also encapsulate/sum over skipped
years - so 2010 is actually the pvf_onm that would exist in 2010 [plus] the
pvf_onm for 2011

Again, the perspective/assumption here is that the operations in the
modeled year persist through gap/unmodeled years -but- in this case we know
what the conditions of future years would be. Said differently, in the
myopic version we assume operations will last 20 years and weight pvf_onm
accordingly; with the intertemporal version we assume they last X years
where X = gap between solve years.

The final, 2050 solve year is weighted to persist for 20 years via the same
methods as the myopic solve -but- we can talk about other terminal year
assumptions that can be made (Barr-Manne, year extensions, ...) if this of
interest to you
